easybuild/easyblocks/i/icc.py

```diff
@@ -105,53 +112,81 @@ def sanity_check_step(self):
     def make_module_req_guess(self):
         """Customize paths to check and add in environment.
         """
+        # New Directory Layout for Intel Parallel Studio XE 2016
+        # https://software.intel.com/en-us/articles/new-directory-layout-for-intel-parallel-studio-xe-2016
+        debuggerpath = 'debugger_%s' % self.version.split('.')[0]
         if self.cfg['m32']:
             # 32-bit toolchain
+            libpaths = ['lib', 'lib/ia32'],
             dirmap = {
                 'PATH': ['bin', 'bin/ia32', 'tbb/bin/ia32'],
-                'LD_LIBRARY_PATH': ['lib', 'lib/ia32'],
-                'LIBRARY_PATH': ['lib', 'lib/ia32'],
+                'LD_LIBRARY_PATH': libpaths,
+                'LIBRARY_PATH': libpaths,
                 'MANPATH': ['man', 'share/man', 'man/en_US'],
                 'IDB_HOME': ['bin/intel64']
             }
         else:
-            # 64-bit toolit
-            dirmap = {
-                'PATH': ['bin', 'bin/intel64', 'tbb/bin/emt64'],
-                'LD_LIBRARY_PATH': ['lib', 'lib/intel64'],
-                'LIBRARY_PATH': ['lib', 'lib/intel64'],
-                'MANPATH': ['man', 'share/man', 'man/en_US'],
-                'IDB_HOME': ['bin/intel64']
-            }
+            if LooseVersion(self.version) < LooseVersion("2016"):
+                # 64-bit toolkit
+                libpaths = ['compiler/lib/intel64', 'lib/intel64', 'debugger/ipt/intel64/lib', 'ipp/lib/intel64', 'tbb/lib/intel64']
```

Comment:
lib exists in the $root, though is a symlink to compilers_and_libraries/linux/lib. Which in its turn is a symlink to  ../../compilers_and_libraries_2016.0.109/linux/compiler/lib
Same for daal, include, ipp, man, mkl and tbb in $root.

This is what comes out of LD_LIBRARY_PATH after sourcing iccvars.sh intel64: (with : replaced by newlines for readability)

.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/compiler/lib/intel64
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/mpi/intel64/lib
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/ipp/../compiler/lib/intel64
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/ipp/lib/intel64
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/compiler/lib/intel64
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/tbb/lib/intel64/gcc4.4
.../icc/2016.0.109/debugger_2016/libipt/intel64/lib
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/daal/lib/intel64_lin
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/daal/../tbb/lib/intel64_lin/gcc4.4
.../icc/2016.0.109/compilers_and_libraries_2016.0.109/linux/daal/../compiler/lib/intel64_lin